activemodel/lib/active_model/validations/associated.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  module Validations
+    class AssociatedValidator < EachValidator # :nodoc:
+      def validate_each(record, attribute, value)
+        if Array(value).reject(&:valid?).any?
```

Comment:
Maybe this could be moved to a method, and the `ActiveRecord` one could inherit from this class and override it.

```suggestion
        if Array(value).reject{ |object| valid_object?(object) }.any?
```

Reply:
I think letting the `ActiveRecord` validator inherit from this one makes sense.
We do the same for `ActiveRecord::Validations::AbsenceValidator`, `ActiveRecord::Validations::LengthValidator`, `ActiveRecord::Validations::PresenceValidator`